app/controllers/users_controller.rb

```diff
@@ -1,11 +1,15 @@
 class UsersController < ApplicationController
-  def show
-    @user = User.find_by id: params[:id]
-    return if @user
-    flash[:none] = t "none"
-    redirect_to root_path
+  before_action :logged_in_user, only: %i(index edit update destroy)
+  before_action :correct_user, only: %i(edit update)
+  before_action :admin_user, only: %i(destroy)
+  before_action :load_user, only: %i(show edit update destroy)
+
+  def index
+    @users = User.page params[:page]
```

Comment:
```suggestion
    @users = User.page(params[:page]).per 10
```
10 cho vào Settings cho anh, đọc mà k hiểu hử?

Reply:
Dạ :'(